### PR TITLE
VIVI-11515 locales support with ytdl

### PR DESCRIPTION
**Pull request by mattd233**

### CODE REVIEW

**Comment by deanylev** [comment]
How do you have conflicts, did you not pull master before checking out?

**Comment by mattd233** [comment]
> How do you have conflicts, did you not pull master before checking out?

Somehow screwed up the rebase I guess

**Comment by deanylev** [comment]
:peppaw: did you mix a rebase and merge?

**Comment by mattd233** [comment]
> :peppaw: did you mix a rebase and merge?

Nope, I think I included an extra commit in my interactive rebase (to squash the commits) which I just left as pick but it shows up here. So I must've done something wrong :piqued:

File: index.js [hunk 1]
```diff
@@ -166,13 +166,13 @@ module.exports.processV2 = (output, origin) => {
   throw 'no url';
 };
 
-module.exports.processV3 = (output, origin) => {
+module.exports.processV3 = (output, origin, locales = []) => {
```
Comment (Alysander) [inline index.js:169]: would it be better to use the en list as a default here?

Comment (mattd233) [reply index.js:169]: I guess you can but I've used the merge approach which is too append the EN_LIST to locales. Doesn't make the difference here but could be helpful when we want to give the user the options.

File: index.js [hunk 2]
```diff
@@ -220,16 +220,19 @@ module.exports.spawnPythonService = (additionalEnv = {}) => {
 
 // private
 
-function findBestSubtitleFile(list) {
+function findBestSubtitleFile(list, locales = []) {
+  // favor locales but some subtitles just have the language code
+  const localesAndLanguages = [...new Set([...locales.map((locale) => locale.toLowerCase()), ...locales.map((locale) => locale.substring(0, 2))])];
```
Comment (Alysander) [inline index.js:225]: Pretty minor but just wondering if we want to preserve the order of the locales? I think it's been implemented as first one is the priority in some situations

**Review by Alysander — APPROVED** [review]

**Comment by Alysander** [comment]
This adds the new locale parameter which defaults to empty array for backwards compatibility. Looks good

**Comment by Alysander** [comment]
I think this would be nice to have in type script in the future and could have a bit more type checking and error handling but that's out of scope for this pr

**Comment by lucashasil** [comment]
> I think this would be nice to have in type script in the future and could have a bit more type checking and error handling but that's out of scope for this pr

Lol don't go and look at all the other lambda repos then

**Comment by deanylev** [comment]
> Lol don't go and look at all the other lambda repos then

only most :peppahands:

**Pull request merged**